tests/integ/kms_utils.py

```diff
@@ -174,19 +174,31 @@ def bucket_with_encryption(boto_session, sagemaker_role):
     bucket_name = "sagemaker-{}-{}-with-kms".format(region, account)
 
     s3 = boto_session.client("s3")
-    try:
-        # 'us-east-1' cannot be specified because it is the default region:
-        # https://github.com/boto/boto3/issues/125
-        if region == "us-east-1":
-            s3.create_bucket(Bucket=bucket_name)
-        else:
-            s3.create_bucket(
-                Bucket=bucket_name, CreateBucketConfiguration={"LocationConstraint": region}
-            )
-
-    except exceptions.ClientError as e:
-        if e.response["Error"]["Code"] != "BucketAlreadyOwnedByYou":
-            raise
+    bucket = boto_session.resource("s3").Bucket(name=bucket_name)
+    if bucket.creation_date is None:
```

Comment:
would it make more sense to move this logic to `s3.py` since (1) we've duplicated the exact same code, and (2) we now have a file just for S3 helpers?

Reply:
Can't drop it in s3 due to a circular dependency, but I went ahead and refactored it into a single method in session. Good callout!